client/state/help/courses/actions.js

```diff
@@ -18,3 +21,29 @@ export function receiveHelpCourses( courses ) {
 		courses
 	};
 }
+
+/**
+ * Returns a function which, when invoked, triggers a network request to fetch
+ * help courses.
+ *
+ * @return {Function}        Action thunk
+ */
+export function requestHelpCourses() {
+	return ( dispatch ) => {
+		dispatch( {
+			type: HELP_COURSES_REQUEST
+		} );
+
+		return wpcom.undocumented().getHelpCourses()
+			.then( courses => {
+				courses.forEach( course => {
+					course.videos.forEach( video => video.parsedDate = Date.parse( video.date ) );
+					course.schedule.forEach( item => item.parsedDate = Date.parse( item.date ) );
+				} );
+				return courses;
+			} )
```

Comment:
This can be made immutable:

```js
import { map } from 'lodash';
const parseDate = x => ( { ...x, parsedDate: Date.parse( x.date ) } );

//...
return wpcom.undocumented().getHelpCourses()
	.then( courses =>
		map( courses, course => ( {
			...course,
			videos: map( course.videos, parseDate ),
			schedule: map( course.schedule, parseDate ),
		} ) )
	)
```